src/neo/SmartContract/ApplicationEngine.Storage.cs

```diff
@@ -94,18 +93,14 @@ private void PutExInternal(StorageContext context, byte[] key, byte[] value, Sto
             StorageItem item = Snapshot.Storages.GetAndChange(skey);
             if (item is null)
             {
-                newDataSize = key.Length + value.Length;
                 Snapshot.Storages.Add(skey, item = new StorageItem());
+                AddGas((key.Length + value.Length) * StoragePrice);
             }
             else
             {
                 if (item.IsConstant) throw new InvalidOperationException();
-                if (value.Length <= item.Value.Length)
-                    newDataSize = 1;
-                else
-                    newDataSize = value.Length - item.Value.Length;
+                AddGas((value.Length / 4 + 1) * StoragePrice);
```

Comment:
Why? If I put 1 byte at first, and then put the real huge data, it can be an attack.

Reply:
Adjusted